Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -148,18 +151,39 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
                 platforms: preprocessInfo.platforms,
                 targetToProducts: preprocessInfo.targetToProducts,
                 targetToResolvedDependencies: preprocessInfo.targetToResolvedDependencies,
+                testTargets: preprocessInfo.testTargets[packageInfo.name] ?? [],
                 targetToModuleMap: preprocessInfo.targetToModuleMap,
                 packageToProject: packageToProject,
                 swiftToolsVersion: swiftToolsVersion
             )
             result[Path(packageInfo.folder.pathString)] = manifest
         }
-
+        
+        let externalDependencies = merge(testTargetsDependencies: preprocessInfo.testTargetsExternalDependencies, with: preprocessInfo.productsToTargetDependencies)
         return DependenciesGraph(
-            externalDependencies: preprocessInfo.productToExternalDependencies,
+            externalDependencies: externalDependencies,
             externalProjects: externalProjects
         )
     }
+    
+    fileprivate func merge(testTargetsDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]], with productsToTargetDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]]) -> [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]] {
+        
+        let platforms: Set<ProjectDescription.Platform> = Set(testTargetsDependencies.keys).union(productsToTargetDependencies.keys)
+        var mergedDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]] = [:]
+        for platform in platforms {
+            let testTargetsToDependencies: [String: [ProjectDescription.TargetDependency]] = testTargetsDependencies[platform] ?? [:]
+            let productTargetsToDependencies: [String: [ProjectDescription.TargetDependency]] = productsToTargetDependencies[platform] ?? [:]
+            let targets: Set<String> = Set(productTargetsToDependencies.keys).union(testTargetsToDependencies.keys)
+            var mergedTargetsToDependencies: [String: [ProjectDescription.TargetDependency]] = [:]
+            for target in targets {
+                let mergedDeps = Set<ProjectDescription.TargetDependency>(productTargetsToDependencies[target] ?? []).union(testTargetsToDependencies[target] ?? [])
+                mergedTargetsToDependencies[target] = Array(mergedDeps)
+            }
+            mergedDependencies[platform] = mergedTargetsToDependencies
+        }
+        
+        return mergedDependencies
+    }
```

Comment:
Can't you do:
```
testTargetsDependencies.merge(productsToTargetDependencies, uniquingKeysWith: {lTargetDependencies, rTargetDependencies in
  lTargetDependencies.merge(rTargetDependencies, uniquingKeysWith: { … })
})
```

Reply:
Also, can there be clashing keys, apart from in the first platform dictionary? 🤔

Reply:
As noted above, all external dependency targets are in existing `productToExternalDependencies` parameter so we don't nee this merge function anymore. Will delete in next commit